pkg/chunk/azure/blob_storage_client.go

```diff
@@ -214,3 +243,41 @@ func (b *BlobStorage) DeleteObject(ctx context.Context, blobID string) error {
 func (b *BlobStorage) PathSeparator() string {
 	return b.delimiter
 }
+
+// Validate the config.
+func (c *BlobStorageConfig) Validate() error {
```

Comment:
Thanks for adding the `Validate()` function. The next step would be calling it. There are two places where it should be called:

1. `Config.Validate()` in `pkg/chunk/storage/factory.go`
2. `RuleStoreConfig.Validate()` in `pkg/ruler/storage.go`